File: src/tools/find_relevant_docs.py
from src.types.api import failure, success


def handle_find_relevant_docs(query: str, top_k: int = 10) -> dict:
    if len(query.strip()) < 2:
        return failure("FIND_QUERY_INVALID", "query должен содержать минимум 2 символа")

    if top_k < 1:
        return failure("FIND_TOPK_INVALID", "top_k должен быть >= 1")

    return success(
        {
            "message": "Этап 1: заглушка find_relevant_docs",
            "query": query,
            "top_k": top_k,
            "results": [],
            "ranking_trace": {},
            "stub": True,
        }
    )

File: src/tools/findRelevantDocs.ts
import { findRelevant } from "../retrieval/hybrid.js";
import { failure, success, type ToolResponse } from "../types/api.js";

export const handleFindRelevantDocs = async (
  query: string,
  topK = 10,
): Promise<ToolResponse<Record<string, unknown>>> => {
  if (query.trim().length < 2) {
    return failure("FIND_QUERY_INVALID", "query должен содержать минимум 2 символа");
  }

  if (topK < 1) {
    return failure("FIND_TOPK_INVALID", "top_k должен быть >= 1");
  }

  const found = await findRelevant(query, topK);

  return success({
    message: "Релевантные чанки найдены",
    query,
    topK,
    ...found,
    stub: false,
  });
};
